<launch>
    <arg name="odom_gazebo_topic_name" default="odom_gazebo" />
    <arg name="odom_publish_topic_name" default="odom_gazebo/converted" />
    <arg name="pub_tf" default="false" />
    <arg name="base_frame" default="base_footprint" />
    <arg name="odom_frame" default="odom" />
    <arg name="add_noise" default="false" />

    <arg name="x" default="0.0"/>
    <arg name="y" default="0.0"/>
    <arg name="z" default="0.0"/>
    <arg name="yaw" default="0.0"/>

    <node name="odom_converter_node" pkg="odom_converter" type="odom_converter_node" output="screen" required="true" args="-x $(arg x) -y $(arg y) -z $(arg z) -yaw $(arg yaw)" >
        <param name="odom_frame" type="string" value="$(arg odom_frame)" />
        <param name="base_frame" type="string" value="$(arg base_frame)" />
        <param name="odom_gazebo_topic_name" type="string" value="$(arg odom_gazebo_topic_name)" />
        <param name="odom_publish_topic_name" type="string" value="$(arg odom_publish_topic_name)" />
        <param name="pub_tf" type="bool" value="$(arg pub_tf)" />
        <param name="add_noise" type="bool" value="$(arg pub_tf)" />
    </node>

</launch>
